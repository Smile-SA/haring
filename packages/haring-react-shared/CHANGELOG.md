# @smile/haring-react-shared

## 0.15.0

### Minor Changes

- [#194](https://github.com/Smile-SA/haring/pull/194) [`97a610a`](https://github.com/Smile-SA/haring/commit/97a610aa02b86423b723ef8ae5d45e8ecf346f92) Thanks [@vapersmile](https://github.com/vapersmile)! - Upgrade mantine version to 7.11.x

## 0.14.0

## 0.13.0

### Minor Changes

- [#178](https://github.com/Smile-SA/haring/pull/178) [`ce8f39a2c52590b2c91950450a84222434913743`](https://github.com/Smile-SA/haring/commit/ce8f39a2c52590b2c91950450a84222434913743) Thanks [@tonai](https://github.com/tonai)! - Rename packages

## 0.12.0

## 0.11.0

## 0.10.0

### Minor Changes

- [#149](https://github.com/Smile-SA/haring/pull/149) [`2706a09`](https://github.com/Smile-SA/haring/commit/2706a097fd86b072cf21d12fe2a97427883671cf) Thanks [@vapersmile](https://github.com/vapersmile)! - Add Calendar style on globlaStyle and add a preview

- [#150](https://github.com/Smile-SA/haring/pull/150) [`34fabfb`](https://github.com/Smile-SA/haring/commit/34fabfbd1a2c048c39adc567548a3ee7e85074ee) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Fixed missing global styles for focus-visible color

## 0.9.0

### Minor Changes

- [`d430e2c`](https://github.com/Smile-SA/haring/commit/d430e2cbcd26871c4db0c53f92a4e4baa6af839c) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Migrated from Mantine V6 to V7

## 0.8.1

## 0.8.0

### Minor Changes

- [#108](https://github.com/Smile-SA/haring/pull/108) [`320634d`](https://github.com/Smile-SA/haring/commit/320634d7ec2048a328cfda1dbcff5d1b70a32563) Thanks [@vapersmile](https://github.com/vapersmile)! - Add custom default style for carousel mantine composent and add 2-Mantine/carousel page

## 0.7.0

### Minor Changes

- [#87](https://github.com/Smile-SA/haring/pull/87) [`dbc06ca`](https://github.com/Smile-SA/haring/commit/dbc06ca55961b69663ab7fdc02609c6525ae361d) Thanks [@tonai](https://github.com/tonai)! - Add theme provider and update barrel file for better next.js compatibility

- [#95](https://github.com/Smile-SA/haring/pull/95) [`5696400`](https://github.com/Smile-SA/haring/commit/5696400e6f703da52db5f7199b50f8251fa76726) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `FilterList` component, renamed `FiltersBar` in `SidebarFilters`
  and `FiltersCheckboxList` in `SearchableCheckboxList`,
  refactored `SearchableCheckboxList`, added common `IFilter` type in shared
  package, added topContent prop to `SwitchableView`, fixed `BrowsingPage` with
  name changes.

## 0.6.0

### Minor Changes

- [#74](https://github.com/Smile-SA/haring/pull/74) [`0ad5377`](https://github.com/Smile-SA/haring/commit/0ad5377535e7c19941da301e8a2ee7298ab70f91) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Refactored actions in `ThumbnailGrid` and `TableGridView` components, moved `typeGuard` helper to shared package, added `typeGuardInterface` function, added tests

## 0.5.0

### Minor Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add dynamic componentProps property in actions

- [#65](https://github.com/Smile-SA/haring/pull/65) [`bd9cfe4`](https://github.com/Smile-SA/haring/commit/bd9cfe42d0b22f5f7f5e7b0de30fdfb22ad3e1c8) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Reworked action/confirm types and moved then in `react-front-kit-shared` then refactored types in `Table`, `Thumbnail` and `ConfirmModal` , Added `ThumbnailGrid` and `TableGridView` components,

- [#68](https://github.com/Smile-SA/haring/pull/68) [`c8b5a39`](https://github.com/Smile-SA/haring/commit/c8b5a3978c8ce7133da0b498d9f0b326f07eb737) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `storybook-pages` package containing example pages, renamed `TestPage`
  into `BrowsingPage` and completed it with integration of `TableGridView`, fixed
  various component styles and props related to page.

### Patch Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add server barrel files

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/haring/pull/53) [`01ee3ea`](https://github.com/Smile-SA/haring/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
