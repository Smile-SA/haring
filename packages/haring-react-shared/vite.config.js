import { resolve } from 'path';

import react from '@vitejs/plugin-react-swc';
import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.tsx'),
      fileName: '[name]',
      formats: ['cjs', 'es'],
    },
    rollupOptions: {
      external: [
        '@mantine/carousel',
        '@mantine/core',
        '@mantine/dates',
        '@mantine/dropzone',
        '@mantine/hooks',
        '@phosphor-icons/react',
        '@smile/haring-react',
        '@smile/haring-react-dates',
        '@smile/haring-react-dropzone',
        '@smile/haring-react-shared',
        '@smile/haring-react-table',
        '@storybook/addon-actions',
        '@storybook/preview-api',
        '@tabler/icons-react',
        '@testing-library/react',
        'dayjs',
        'embla-carousel-react',
        'mantine-react-table',
        'react',
        'react/jsx-runtime',
      ],
      input: {
        index: resolve(__dirname, 'src/index.tsx'),
        server: resolve(__dirname, 'src/server.tsx'),
        'storybook-utils': resolve(__dirname, 'src/storybook-utils/index.ts'),
        'test-utils': resolve(__dirname, 'src/test-utils/index.ts'),
      },
      output: {
        banner: (chunkInfo) => {
          if (
            ['src/index.tsx'].find((modulePath) =>
              chunkInfo.facadeModuleId?.endsWith(modulePath),
            )
          ) {
            return `"use client"`;
          }
          return '';
        },
        preserveModules: false,
      },
    },
  },
  plugins: [
    react(),
    dts({
      insertTypesEntry: true,
      tsconfigPath: './tsconfig.build.json',
    }),
  ],
});
