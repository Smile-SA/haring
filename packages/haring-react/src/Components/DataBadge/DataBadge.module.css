.root {
  border-radius: 8px;
  padding: 16px 24px;
  font-size: 14px;
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.rootMd {
  padding: 6px 10px;
}

.groupRoot {
  display: flex;
  white-space: normal;
  display: flex;
}

.number {
  margin: auto 0;
  margin-right: 16px;
  display: block;
  font-size: 34px;
  min-width: fit-content;
}

.numberMd {
  font-size: 20px;
  margin-right: 12px;
}

.text {
  margin: 0;
  max-width: 123px;
}

.textMd {
  font-weight: 400;
  text-transform: lowercase;
  max-width: 100%;
}
