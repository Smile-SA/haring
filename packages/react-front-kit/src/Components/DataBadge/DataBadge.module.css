.root {
  border-radius: 8px;
  padding: 16px 24px;
  font-size: 14px;
  display: block;
  height: auto;
  width: auto;
}

.rootMd {
  padding: auto;
}

.groupRoot {
  display: flex;
  white-space: wrap;
  display: flex;
}

.number {
  margin: auto 0;
  margin-right: 16px;
  display: block;
  font-size: 34px;
}

.numberMd {
  font-size: 20px;
  margin-right: 12px;
}

.text {
  margin: 0;
  max-width: 123px;
}

.textMd {
  font-weight: 400;
  text-transform: lowercase;
  max-width: 100%;
}
