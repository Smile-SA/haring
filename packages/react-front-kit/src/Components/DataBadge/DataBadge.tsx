'use client';

import type { BadgeProps, DefaultMantineColor } from '@mantine/core';
import type { ReactElement } from 'react';

import { Badge, getThemeColor, useMantineTheme } from '@mantine/core';

import classes from './DataBadge.module.css';

type ISize = 'lg' | 'md';

export interface IDataBadgeProps extends Omit<BadgeProps, 'color'> {
  children?: string;
  color?: DefaultMantineColor;
  number?: number;
  numberSize?: 1 | 10 | 100 | 1000;
  size?: ISize;
}

export function DataBadge(props: IDataBadgeProps): ReactElement {
  const {
    children,
    color,
    number = 10,
    size = 'md',
    numberSize = 10,
    ...badgeProps
  } = props;

  const theme = useMantineTheme();

  function getNumberSize(size: ISize): Record<string, string> {
    let value = '40px';

    if (size === 'md') {
      switch (numberSize) {
        case 1:
          value = '12px';
          break;
        case 10:
          value = '24px';
          break;
        case 100:
          value = '35px';
          break;
        case 1000:
          value = '47px';
          break;
      }
    } else {
      switch (numberSize) {
        case 1:
          value = '20px';
          break;
        case 100:
          value = '60px';
          break;
        case 1000:
          value = '80px';
          break;
      }
    }

    return { width: value };
  }

  return (
    <Badge
      autoContrast
      bg={!color ? 'transparent' : undefined}
      className={[classes.root, size === 'md' && classes.rootMd].join(' ')}
      color={color ? color : getThemeColor(theme.colors.dark[3], theme)}
      variant="light"
      {...badgeProps}
    >
      <div className={classes.groupRoot}>
        <span
          className={[classes.number, size === 'md' && classes.numberMd].join(
            ' ',
          )}
          style={getNumberSize(size)}
        >
          {number}
        </span>
        <p
          className={[classes.text, size === 'md' && classes.textMd].join(' ')}
        >
          {children}
        </p>
      </div>
    </Badge>
  );
}
