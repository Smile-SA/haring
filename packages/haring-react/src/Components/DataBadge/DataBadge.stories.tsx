import type { Meta, StoryObj } from '@storybook/react';

import { DEFAULT_THEME } from '@mantine/core';

import { DataBadge as Cmp } from './DataBadge';

const colorOptions = Object.keys(DEFAULT_THEME.colors);
const meta = {
  argTypes: {
    color: {
      control: 'select',
      options: colorOptions,
    },
  },
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/DataBadge',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const DataBadgeLg: IStory = {
  args: {
    children: 'Application in progress',
    color: 'cyan',
    number: 48,
    size: 'lg',
  },
};

export const DataBadgeMd: IStory = {
  args: {
    children: 'Application in progress',
    number: 18,
  },
};
