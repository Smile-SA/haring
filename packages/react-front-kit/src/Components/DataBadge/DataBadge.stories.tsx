import type { Meta, StoryObj } from '@storybook/react';

import { DataBadge as Cmp } from './DataBadge';

const meta = {
  argTypes: {
    color: {
      control: { type: 'color' },
    },
  },
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/DataBadge',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const DataBadgeLg: IStory = {
  args: {
    children: 'Application in progress',
    color: 'cyan',
    number: 48,
    size: 'lg',
  },
};

export const DataBadgeMd: IStory = {
  args: {
    children: 'Application in progress',
    number: 18,
  },
};
