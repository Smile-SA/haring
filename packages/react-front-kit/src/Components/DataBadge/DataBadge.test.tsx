import { renderWithProviders } from '@smile/haring-react-shared/test-utils';

import { DataBadge } from './DataBadge';

describe('DataBadge', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(<DataBadge />);
    expect(container).toMatchSnapshot();
  });
});
