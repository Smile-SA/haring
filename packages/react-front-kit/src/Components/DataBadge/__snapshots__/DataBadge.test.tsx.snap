// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataBadge matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-cursor-type: pointer;--mantine-color-black: #2e2e2e;--mantine-font-family: var(--rfk-font, Open Sans);--mantine-font-family-headings: var(--rfk-font, Open Sans);--mantine-radius-default: calc(1.5rem * var(--mantine-scale));--mantine-primary-color-filled: var(--mantine-color-cyan-filled);--mantine-primary-color-filled-hover: var(--mantine-color-cyan-filled-hover);--mantine-primary-color-light: var(--mantine-color-cyan-light);--mantine-primary-color-light-hover: var(--mantine-color-cyan-light-hover);--mantine-primary-color-light-color: var(--mantine-color-cyan-light-color);--mantine-font-size-xs: 10px;--mantine-font-size-sm: 12px;--mantine-font-size-md: 14px;--mantine-font-size-lg: 18px;--mantine-font-size-xl: 20px;--mantine-primary-color-0: var(--mantine-color-cyan-0);--mantine-primary-color-1: var(--mantine-color-cyan-1);--mantine-primary-color-2: var(--mantine-color-cyan-2);--mantine-primary-color-3: var(--mantine-color-cyan-3);--mantine-primary-color-4: var(--mantine-color-cyan-4);--mantine-primary-color-5: var(--mantine-color-cyan-5);--mantine-primary-color-6: var(--mantine-color-cyan-6);--mantine-primary-color-7: var(--mantine-color-cyan-7);--mantine-primary-color-8: var(--mantine-color-cyan-8);--mantine-primary-color-9: var(--mantine-color-cyan-9);--mantine-h1-font-size: 26px;--mantine-h1-line-height: 135%;--mantine-h2-font-size: 18px;--mantine-h2-line-height: 155%;--mantine-h3-font-size: 14px;--mantine-h3-line-height: 155%;}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-cyan-4);--mantine-color-dark-filled: var(--mantine-color-dark-9);--mantine-color-dark-filled-hover: var(--mantine-color-dark-8);--mantine-color-dark-light: rgba(36, 36, 36, 0.15);--mantine-color-dark-light-hover: rgba(36, 36, 36, 0.2);--mantine-color-dark-light-color: var(--mantine-color-dark-4);--mantine-color-dark-outline: var(--mantine-color-dark-5);--mantine-color-dark-outline-hover: rgba(59, 59, 59, 0.05);--mantine-color-gray-filled: var(--mantine-color-gray-9);--mantine-color-gray-filled-hover: var(--mantine-color-gray-8);--mantine-color-gray-light: rgba(73, 80, 87, 0.15);--mantine-color-gray-light-hover: rgba(73, 80, 87, 0.2);--mantine-color-gray-light-color: var(--mantine-color-gray-4);--mantine-color-gray-outline: var(--mantine-color-gray-5);--mantine-color-gray-outline-hover: rgba(173, 181, 189, 0.05);--mantine-color-red-filled: var(--mantine-color-red-9);--mantine-color-red-filled-hover: var(--mantine-color-red-8);--mantine-color-red-light: rgba(240, 62, 62, 0.15);--mantine-color-red-light-hover: rgba(240, 62, 62, 0.2);--mantine-color-red-light-color: var(--mantine-color-red-4);--mantine-color-red-outline: var(--mantine-color-red-5);--mantine-color-red-outline-hover: rgba(255, 107, 107, 0.05);--mantine-color-pink-filled: var(--mantine-color-pink-9);--mantine-color-pink-filled-hover: var(--mantine-color-pink-8);--mantine-color-pink-light: rgba(214, 51, 108, 0.15);--mantine-color-pink-light-hover: rgba(214, 51, 108, 0.2);--mantine-color-pink-light-color: var(--mantine-color-pink-4);--mantine-color-pink-outline: var(--mantine-color-pink-5);--mantine-color-pink-outline-hover: rgba(240, 101, 149, 0.05);--mantine-color-grape-filled: var(--mantine-color-grape-9);--mantine-color-grape-filled-hover: var(--mantine-color-grape-8);--mantine-color-grape-light: rgba(174, 62, 201, 0.15);--mantine-color-grape-light-hover: rgba(174, 62, 201, 0.2);--mantine-color-grape-light-color: var(--mantine-color-grape-4);--mantine-color-grape-outline: var(--mantine-color-grape-5);--mantine-color-grape-outline-hover: rgba(204, 93, 232, 0.05);--mantine-color-violet-filled: var(--mantine-color-violet-9);--mantine-color-violet-filled-hover: var(--mantine-color-violet-8);--mantine-color-violet-light: rgba(112, 72, 232, 0.15);--mantine-color-violet-light-hover: rgba(112, 72, 232, 0.2);--mantine-color-violet-light-color: var(--mantine-color-violet-4);--mantine-color-violet-outline: var(--mantine-color-violet-5);--mantine-color-violet-outline-hover: rgba(132, 94, 247, 0.05);--mantine-color-indigo-filled: var(--mantine-color-indigo-9);--mantine-color-indigo-filled-hover: var(--mantine-color-indigo-8);--mantine-color-indigo-light: rgba(66, 99, 235, 0.15);--mantine-color-indigo-light-hover: rgba(66, 99, 235, 0.2);--mantine-color-indigo-light-color: var(--mantine-color-indigo-4);--mantine-color-indigo-outline: var(--mantine-color-indigo-5);--mantine-color-indigo-outline-hover: rgba(92, 124, 250, 0.05);--mantine-color-blue-filled: var(--mantine-color-blue-9);--mantine-color-blue-filled-hover: var(--mantine-color-blue-8);--mantine-color-blue-light: rgba(28, 126, 214, 0.15);--mantine-color-blue-light-hover: rgba(28, 126, 214, 0.2);--mantine-color-blue-light-color: var(--mantine-color-blue-4);--mantine-color-blue-outline: var(--mantine-color-blue-5);--mantine-color-blue-outline-hover: rgba(51, 154, 240, 0.05);--mantine-color-cyan-filled: var(--mantine-color-cyan-9);--mantine-color-cyan-filled-hover: var(--mantine-color-cyan-8);--mantine-color-cyan-light: rgba(16, 152, 173, 0.15);--mantine-color-cyan-light-hover: rgba(16, 152, 173, 0.2);--mantine-color-cyan-light-color: var(--mantine-color-cyan-4);--mantine-color-cyan-outline: var(--mantine-color-cyan-5);--mantine-color-cyan-outline-hover: rgba(34, 184, 207, 0.05);--mantine-color-teal-filled: var(--mantine-color-teal-9);--mantine-color-teal-filled-hover: var(--mantine-color-teal-8);--mantine-color-teal-light: rgba(12, 166, 120, 0.15);--mantine-color-teal-light-hover: rgba(12, 166, 120, 0.2);--mantine-color-teal-light-color: var(--mantine-color-teal-4);--mantine-color-teal-outline: var(--mantine-color-teal-5);--mantine-color-teal-outline-hover: rgba(32, 201, 151, 0.05);--mantine-color-green-filled: var(--mantine-color-green-9);--mantine-color-green-filled-hover: var(--mantine-color-green-8);--mantine-color-green-light: rgba(55, 178, 77, 0.15);--mantine-color-green-light-hover: rgba(55, 178, 77, 0.2);--mantine-color-green-light-color: var(--mantine-color-green-4);--mantine-color-green-outline: var(--mantine-color-green-5);--mantine-color-green-outline-hover: rgba(81, 207, 102, 0.05);--mantine-color-lime-filled: var(--mantine-color-lime-9);--mantine-color-lime-filled-hover: var(--mantine-color-lime-8);--mantine-color-lime-light: rgba(116, 184, 22, 0.15);--mantine-color-lime-light-hover: rgba(116, 184, 22, 0.2);--mantine-color-lime-light-color: var(--mantine-color-lime-4);--mantine-color-lime-outline: var(--mantine-color-lime-5);--mantine-color-lime-outline-hover: rgba(148, 216, 45, 0.05);--mantine-color-yellow-filled: var(--mantine-color-yellow-9);--mantine-color-yellow-filled-hover: var(--mantine-color-yellow-8);--mantine-color-yellow-light: rgba(245, 159, 0, 0.15);--mantine-color-yellow-light-hover: rgba(245, 159, 0, 0.2);--mantine-color-yellow-light-color: var(--mantine-color-yellow-4);--mantine-color-yellow-outline: var(--mantine-color-yellow-5);--mantine-color-yellow-outline-hover: rgba(252, 196, 25, 0.05);--mantine-color-orange-filled: var(--mantine-color-orange-9);--mantine-color-orange-filled-hover: var(--mantine-color-orange-8);--mantine-color-orange-light: rgba(247, 103, 7, 0.15);--mantine-color-orange-light-hover: rgba(247, 103, 7, 0.2);--mantine-color-orange-light-color: var(--mantine-color-orange-4);--mantine-color-orange-outline: var(--mantine-color-orange-5);--mantine-color-orange-outline-hover: rgba(255, 146, 43, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-text: #2e2e2e;--mantine-color-anchor: var(--mantine-color-cyan-9);--mantine-color-dark-filled: var(--mantine-color-dark-9);--mantine-color-dark-filled-hover: var(--mantine-color-dark-8);--mantine-color-dark-light: rgba(20, 20, 20, 0.1);--mantine-color-dark-light-hover: rgba(20, 20, 20, 0.12);--mantine-color-dark-light-color: var(--mantine-color-dark-9);--mantine-color-dark-outline: var(--mantine-color-dark-9);--mantine-color-dark-outline-hover: rgba(20, 20, 20, 0.05);--mantine-color-gray-filled: var(--mantine-color-gray-9);--mantine-color-gray-filled-hover: var(--mantine-color-gray-8);--mantine-color-gray-light: rgba(33, 37, 41, 0.1);--mantine-color-gray-light-hover: rgba(33, 37, 41, 0.12);--mantine-color-gray-light-color: var(--mantine-color-gray-9);--mantine-color-gray-outline: var(--mantine-color-gray-9);--mantine-color-gray-outline-hover: rgba(33, 37, 41, 0.05);--mantine-color-red-filled: var(--mantine-color-red-9);--mantine-color-red-filled-hover: var(--mantine-color-red-8);--mantine-color-red-light: rgba(201, 42, 42, 0.1);--mantine-color-red-light-hover: rgba(201, 42, 42, 0.12);--mantine-color-red-light-color: var(--mantine-color-red-9);--mantine-color-red-outline: var(--mantine-color-red-9);--mantine-color-red-outline-hover: rgba(201, 42, 42, 0.05);--mantine-color-pink-filled: var(--mantine-color-pink-9);--mantine-color-pink-filled-hover: var(--mantine-color-pink-8);--mantine-color-pink-light: rgba(166, 30, 77, 0.1);--mantine-color-pink-light-hover: rgba(166, 30, 77, 0.12);--mantine-color-pink-light-color: var(--mantine-color-pink-9);--mantine-color-pink-outline: var(--mantine-color-pink-9);--mantine-color-pink-outline-hover: rgba(166, 30, 77, 0.05);--mantine-color-grape-filled: var(--mantine-color-grape-9);--mantine-color-grape-filled-hover: var(--mantine-color-grape-8);--mantine-color-grape-light: rgba(134, 46, 156, 0.1);--mantine-color-grape-light-hover: rgba(134, 46, 156, 0.12);--mantine-color-grape-light-color: var(--mantine-color-grape-9);--mantine-color-grape-outline: var(--mantine-color-grape-9);--mantine-color-grape-outline-hover: rgba(134, 46, 156, 0.05);--mantine-color-violet-filled: var(--mantine-color-violet-9);--mantine-color-violet-filled-hover: var(--mantine-color-violet-8);--mantine-color-violet-light: rgba(95, 61, 196, 0.1);--mantine-color-violet-light-hover: rgba(95, 61, 196, 0.12);--mantine-color-violet-light-color: var(--mantine-color-violet-9);--mantine-color-violet-outline: var(--mantine-color-violet-9);--mantine-color-violet-outline-hover: rgba(95, 61, 196, 0.05);--mantine-color-indigo-filled: var(--mantine-color-indigo-9);--mantine-color-indigo-filled-hover: var(--mantine-color-indigo-8);--mantine-color-indigo-light: rgba(54, 79, 199, 0.1);--mantine-color-indigo-light-hover: rgba(54, 79, 199, 0.12);--mantine-color-indigo-light-color: var(--mantine-color-indigo-9);--mantine-color-indigo-outline: var(--mantine-color-indigo-9);--mantine-color-indigo-outline-hover: rgba(54, 79, 199, 0.05);--mantine-color-blue-filled: var(--mantine-color-blue-9);--mantine-color-blue-filled-hover: var(--mantine-color-blue-8);--mantine-color-blue-light: rgba(24, 100, 171, 0.1);--mantine-color-blue-light-hover: rgba(24, 100, 171, 0.12);--mantine-color-blue-light-color: var(--mantine-color-blue-9);--mantine-color-blue-outline: var(--mantine-color-blue-9);--mantine-color-blue-outline-hover: rgba(24, 100, 171, 0.05);--mantine-color-cyan-filled: var(--mantine-color-cyan-9);--mantine-color-cyan-filled-hover: var(--mantine-color-cyan-8);--mantine-color-cyan-light: rgba(11, 114, 133, 0.1);--mantine-color-cyan-light-hover: rgba(11, 114, 133, 0.12);--mantine-color-cyan-light-color: var(--mantine-color-cyan-9);--mantine-color-cyan-outline: var(--mantine-color-cyan-9);--mantine-color-cyan-outline-hover: rgba(11, 114, 133, 0.05);--mantine-color-teal-filled: var(--mantine-color-teal-9);--mantine-color-teal-filled-hover: var(--mantine-color-teal-8);--mantine-color-teal-light: rgba(8, 127, 91, 0.1);--mantine-color-teal-light-hover: rgba(8, 127, 91, 0.12);--mantine-color-teal-light-color: var(--mantine-color-teal-9);--mantine-color-teal-outline: var(--mantine-color-teal-9);--mantine-color-teal-outline-hover: rgba(8, 127, 91, 0.05);--mantine-color-green-filled: var(--mantine-color-green-9);--mantine-color-green-filled-hover: var(--mantine-color-green-8);--mantine-color-green-light: rgba(43, 138, 62, 0.1);--mantine-color-green-light-hover: rgba(43, 138, 62, 0.12);--mantine-color-green-light-color: var(--mantine-color-green-9);--mantine-color-green-outline: var(--mantine-color-green-9);--mantine-color-green-outline-hover: rgba(43, 138, 62, 0.05);--mantine-color-lime-filled: var(--mantine-color-lime-9);--mantine-color-lime-filled-hover: var(--mantine-color-lime-8);--mantine-color-lime-light: rgba(92, 148, 13, 0.1);--mantine-color-lime-light-hover: rgba(92, 148, 13, 0.12);--mantine-color-lime-light-color: var(--mantine-color-lime-9);--mantine-color-lime-outline: var(--mantine-color-lime-9);--mantine-color-lime-outline-hover: rgba(92, 148, 13, 0.05);--mantine-color-yellow-filled: var(--mantine-color-yellow-9);--mantine-color-yellow-filled-hover: var(--mantine-color-yellow-8);--mantine-color-yellow-light: rgba(230, 119, 0, 0.1);--mantine-color-yellow-light-hover: rgba(230, 119, 0, 0.12);--mantine-color-yellow-light-color: var(--mantine-color-yellow-9);--mantine-color-yellow-outline: var(--mantine-color-yellow-9);--mantine-color-yellow-outline-hover: rgba(230, 119, 0, 0.05);--mantine-color-orange-filled: var(--mantine-color-orange-9);--mantine-color-orange-filled-hover: var(--mantine-color-orange-8);--mantine-color-orange-light: rgba(217, 72, 15, 0.1);--mantine-color-orange-light-hover: rgba(217, 72, 15, 0.12);--mantine-color-orange-light-color: var(--mantine-color-orange-9);--mantine-color-orange-outline: var(--mantine-color-orange-9);--mantine-color-orange-outline-hover: rgba(217, 72, 15, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="  m_347db0ec mantine-Badge-root"
    data-variant="light"
    style="--badge-bg: rgba(105, 105, 105, 0.1); --badge-color: #696969; --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; background: transparent;"
  >
    <span
      class="m_5add502a mantine-Badge-label"
    >
      <div>
        <span
          class=" "
        />
        <p
          class=" "
        />
      </div>
    </span>
  </div>
</div>
`;
